internal/openfga/interface.go

```diff
@@ -0,0 +1,37 @@
+package openfga
+
+import (
+	"io/ioutil"
+
+	"github.com/cloudoperators/heureka/internal/util"
+	"github.com/sirupsen/logrus"
+)
+
+type Authorization interface {
+	// check if userId has permission on resourceId
+	CheckPermission(userId string, resourceId string, resourceType string, permission string) (bool, error)
+	// add relationship between userId and resourceId
+	AddRelation(userId string, resourceId string, resourceType string, relation string) error
```

Comment:
Same for AddRelation/RemoveRelation. I suggest using a type:
```
type Relation struct {
    SubjectType: SubjectType
    SubjectId: string
    Relation: Relation
    ObjectType ObjectType
    ObjectId: string
}
```

Looking at the attributes, it looks the same as the `PermissionCheck` but we might add different methods, so I'd use 2 different types.